src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -209,17 +217,18 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
                             {
                                 _logger.LogWarning(ex, "Removing errored webhost language worker channel for runtime: {workerRuntime} workerId:{workerId}", _workerRuntime, workerId);
                                 await _webHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(_workerRuntime, workerId, ex);
-                                InitializeWebhostLanguageWorkerChannel();
                             }
                         }
                     }
                     StartWorkerProcesses(webhostLanguageWorkerChannels.Count(), InitializeWebhostLanguageWorkerChannel);
-                    SetFunctionDispatcherStateToInitializedAndLog();
+                    if (webhostLanguageWorkerChannels.Count() > 0)
+                    {
+                        SetFunctionDispatcherStateToInitializedAndLog();
+                    }
                 }
                 else
                 {
-                    await InitializeJobhostLanguageWorkerChannelAsync(0);
-                    StartWorkerProcesses(1, InitializeJobhostLanguageWorkerChannelAsync);
+                    StartWorkerProcesses(0, InitializeJobhostLanguageWorkerChannelAsync);
```

Comment:
This can potentially improve cold start for non-placeholder scenarios as we are not waiting for process to start.

Reply:
For the first process - we do want to block. Launching multiple  processes can be "fire and forget".
As good way to test this E2E - locally - Set up a queue with 100+ messages then start the functions runtime - check if executions succeed.

Reply:
I tested this with 100+ messages and all executions were succeed.
In old code we also set up `RpcFunctionInvocationDispatcher.State=FunctionInvocationDispatcherState.Initialized`; state in the non-blocking code:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L119

Reply:
Thanks for confirming!